### DISCUSSION TITLE
Using Pekko with Application Insights for logs and telemetry

### DISCUSSION THREAD

**Discussion by AnastasiiaZu** [apache/pekko#1031] [Q&A] [answered]
Hi! 

My company is currently experimenting with Pekko 1.0.0. We collect our logs via Microsoft's Application Insights. Java logging agent from Application Insights [provides](https://learn.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-config#preview-instrumentations) an Akka instrumentation as an option:
```
    "instrumentation": {
      "akka": {
        "enabled": true
      },
```
 but not Pekko at the moment. 
As a result, when Pekko is used, no logs are being published to AI.

Do you know of a way to solve this problem? I saw that there was an auto-instrumentation solution for Open Telemetry, but not for AI.

Best,
Anastasiia Zu

**Comment by He-Pin** [comment]
If it's closed source, you may have to send an request to MS...

**Reply by AnastasiiaZu** [reply]
The code to this feature can be found [here](https://github.com/microsoft/ApplicationInsights-Java/blob/bd91f620189d797bee62541a532740d25c8c46e4/agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java#L476)

**Comment by pjfanning** [comment, marked as answer]
It is my understanding that Lightbend have a paid-for solution. We can't open source software that has a non open source license. Approach your vendor and ask them for a solution. If you are using an open source telemetry - give us the details so we look at it to see if it can be updated to support Pekko.

**Reply by He-Pin** [reply]
It's ai from MSFT, you could asked they from the support console to adopt pekko, that will help pekko too.

**Reply by pjfanning** [reply]
I still don't understand what the OP expects us to do about a commercial project - why don't they contact their commercial support?

**Reply by pjfanning** [reply]
seems like a vert similar issue to the Dynatrace discussion - https://github.com/apache/incubator-pekko/discussions/505

**Reply by He-Pin** [reply]
I think he may just not sure who provides  that feature. 

But anyway, he/she is a maybe pekko user

**Reply by AnastasiiaZu** [reply]
Hi @pjfanning, I opened an [issue](https://github.com/microsoft/ApplicationInsights-Java/issues/3512) with Microsoft as well, will let you know how it goes

**Reply by pjfanning** [reply]
Thanks @AnastasiiaZu - just to cross-post here but it looks like Microsoft lib just wraps https://github.com/open-telemetry/opentelemetry-java-instrumentation and that lib supports Pekko. So hopefully, the MS lib can be easily changed to support Pekko too.
